### PR TITLE
Add Crossfeed integration and api.integration records (A, AAAA, TXT, CAA)

### PR DESCRIPTION
**Pull request by cduhn17**
Update domain name for integration

## 🗣 Description ##

We need to add the api.integration domain name and issue letsencrypt ssl certificates. There is a need for both a TXT record and CAA record and adding AAAA and A records for integration.

## 💭 Motivation and context ##

There are no existing ACME certs

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by cduhn17** [comment]
@dav3r , I have some code commented out that will be a follow-on PR, I have this commented out because I have to have the SSL cert in place before I can get the values from AWS for the AAAA and A records for api.integration.crossfeed.cyber.dhs.gov

**Comment by dav3r** [comment]
> @dav3r , I have some code commented out that will be a follow-on PR, I have this commented out because I have to have the SSL cert in place before I can get the values from AWS for the AAAA and A records for api.integration.crossfeed.cyber.dhs.gov

Please remove the commented-out code since it's not relevant to this PR - it should be added in your next PR.

**Review by jsf9k — APPROVED** [review]

File: route53_crossfeed_app.tf [hunk 1]
```diff
@@ -337,6 +337,84 @@ resource "aws_route53_record" "crossfeed_integration_digicert_letsencrypt_CAA" {
...
+#   alias {
+#     name                   = "d-y5130perp8.execute-api.us-east-1.amazonaws.com."
+#     evaluate_target_health = false
+#     zone_id                = var.api_gateway_zone_id
+#   }
+#   name    = "api.integration.crossfeed.${aws_route53_zone.cyber_dhs_gov.name}"
+#   type    = "AAAA"
+#   zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
+# }
+
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: Add a section header for integration API entries:
```suggestion

# ------------------------------------------------------------------------------
# Integration API entries
# ------------------------------------------------------------------------------

```

**Comment by jsf9k** [comment]
@cduhn17 - When you make a change please ensure that the commit message speaks to the change that was made.  "Update route53_crossfeed_app.tf" is very generic and tells a future developer nothing about the content of the commit.

**Comment by cduhn17** [comment]
@dav3r am I GTG after a tf apply? Thanks

File: route53_crossfeed_app.tf [hunk 2]
```diff
@@ -337,6 +337,88 @@ resource "aws_route53_record" "crossfeed_integration_digicert_letsencrypt_CAA" {
...
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: Remove unused code:
```suggestion
```

**Comment by cduhn17** [comment]
@dav3r My intent with the commented out code was that the code will be added right after this PR is applied. But i removed it for now. Thanks

**Comment by dav3r** [comment]
> @dav3r My intent with the commented out code was that the code will be added right after this PR is applied. But i removed it for now. Thanks

Sorry, that wasn't clear before - I thought you meant that code was going to be added in a subsequent PR.

**Comment by cduhn17** [comment]
Let me know when i can try the verification. Thanks

**Comment by dav3r** [comment]
> Let me know when i can try the verification. Thanks

The changes have been applied - give it a shot.

**Comment by cduhn17** [comment]
Thanks all i have the certificates in hand

File: route53_crossfeed_app.tf [hunk 3]
```diff
@@ -337,6 +337,88 @@ resource "aws_route53_record" "crossfeed_integration_digicert_letsencrypt_CAA" {
...
+    name                   = "d-sl8v3ffkci.execute-api.us-east-1.amazonaws.com."
+    evaluate_target_health = false
+    zone_id                = var.api_gateway_zone_id
+  }
+  name    = "integration.crossfeed.${aws_route53_zone.cyber_dhs_gov.name}"
+  type    = "A"
+  zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
+}
+
+# The following A and AAAA will follow once we have the SSL certificate for the API integration environment
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: This comment is no longer needed.
```suggestion
```

File: route53_crossfeed_app.tf [hunk 4]
```diff
@@ -337,6 +337,87 @@ resource "aws_route53_record" "crossfeed_integration_digicert_letsencrypt_CAA" {
...
+
+  alias {
+    name                   = "d-3kn94fjcb8.execute-api.us-east-1.amazonaws.com."
+    evaluate_target_health = false
+    zone_id                = var.api_gateway_zone_id
+  }
+  name    = "api.integration.crossfeed.${aws_route53_zone.cyber_dhs_gov.name}"
+  type    = "AAAA"
+  zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
+}
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: These two resource blocks should be moved down into the "Integration API entries" section and placed to preserve alphabetical ordering by resource name.

File: route53_crossfeed_app.tf [hunk 5]
```diff
@@ -337,6 +337,87 @@ resource "aws_route53_record" "crossfeed_integration_digicert_letsencrypt_CAA" {
...
+
+  alias {
+    name                   = "d-sl8v3ffkci.execute-api.us-east-1.amazonaws.com."
+    evaluate_target_health = false
+    zone_id                = var.api_gateway_zone_id
+  }
+  name    = "integration.crossfeed.${aws_route53_zone.cyber_dhs_gov.name}"
+  type    = "A"
+  zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
+}
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: This resource block should be moved up to preserve alphabetical ordering by resource name in the "Integration entries" section.

File: route53_crossfeed_app.tf [hunk 6]
```diff
@@ -337,6 +337,87 @@ resource "aws_route53_record" "crossfeed_integration_digicert_letsencrypt_CAA" {
...
+
+  alias {
+    name                   = "d-sl8v3ffkci.execute-api.us-east-1.amazonaws.com."
+    evaluate_target_health = false
+    zone_id                = var.api_gateway_zone_id
+  }
+  name    = "integration.crossfeed.${aws_route53_zone.cyber_dhs_gov.name}"
+  type    = "AAAA"
+  zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
+}
```
Comment (dav3r) [inline route53_crossfeed_app.tf]: This resource block should be moved up to preserve alphabetical ordering by resource name in the "Integration entries" section.

**Review by dav3r — CHANGES_REQUESTED** [review]

File: route53_crossfeed_app.tf (hunk 5 above)
Comment (cduhn17) [reply route53_crossfeed_app.tf]: Moved order to alphabetize record type per commit 
[4071b5b](https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/113/commits/4071b5be643565b9ce3a896931f5a7db24de79de)

File: route53_crossfeed_app.tf (hunk 6 above)
Comment (cduhn17) [reply route53_crossfeed_app.tf]: Moved order to alphabetize record type per commit 
[4071b5b](https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/113/commits/4071b5be643565b9ce3a896931f5a7db24de79de)

**Comment by dav3r** [comment]
@cduhn17 I applied the current Terraform and created the integration API A and AAA records.  Please let me know if there is anything else planned in this PR.

**Comment by cduhn17** [comment]
@dav3r , Thanks for all of your help. There is nothing further for this PR

**Review by dav3r — APPROVED** [review]
LGTM

File: route53_crossfeed_app.tf (hunk 4 above)
Comment (dav3r) [reply route53_crossfeed_app.tf]: This was handled in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/113/commits/4071b5be643565b9ce3a896931f5a7db24de79de.

**Pull request merged**